src/main/java/net/minecraftforge/common/data/GlobalLootModifierProvider.java

```diff
@@ -91,11 +87,17 @@ public void run(CachedOutput cache) throws IOException
      * Passes in the data needed to create the file without any extra objects.
      *
      * @param modifier      The name of the modifier, which will be the file name.
-     * @param serializer    The serializer of this modifier.
+     * @param serializer    The registered serializer of this modifier.
+     * @param instance      The instance to serialize
      */
-    public <T extends IGlobalLootModifier> void add(String modifier, GlobalLootModifierSerializer<T> serializer, T instance)
+    public <T extends IGlobalLootModifier> void add(String modifier, Codec<T> serializer, T instance)
     {
-        this.toSerialize.put(modifier, new Tuple<>(serializer, serializer.write(instance)));
+        JsonElement element = serializer.encodeStart(JsonOps.INSTANCE, instance).getOrThrow(false, s -> {});
+        if (!(element instanceof JsonObject obj))
+            throw new RuntimeException("Expecting loot modifier to be a Json Object!");
+        String serializerName = ForgeRegistries.LOOT_MODIFIER_SERIALIZERS.get().getKey(serializer).toString();
+        obj.addProperty("type", serializerName);
+        this.toSerialize.put(modifier, obj);
```

Comment:
Aren't there dispatch codecs for this kind of stuff?

Reply:
indeed, mb